### PR TITLE
sanitizeQuery depthLimit, lengthLimit, blockListing functionality | jest unit testing for sanitizeQuery and createSecrets

### PR DESCRIPTION
**Pull request by rscalderon**

### CODE REVIEW

**Comment by sifulsidd** [comment]
BEAUTIFUL!

File: sanitize.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  // if input length exceeds passed-in param length, throw new Error
+  if (input.length > length)
+    throw new Error(
+      'Maximum query length exceeded, modify query or maximum permitted query length'
+    );
+  // if input depth is within the permitted limits, return input
+  return input;
+};
+
+// console.log(sanitize('hello'));
```
Comment (joie-zhang) [inline sanitize.js:81]: Remove when you get the chance!

File: sanitizeQuery.js [hunk 2]
```diff
@@ -1,85 +1,47 @@
...
-// init helper func deepLimit that accepts two params, input (a JSON string graphQL query) and length (a number value) and throws an error if the passed-in input has a length greater than length, else returns input
-const lengthLimit = (input, length) => {
-  // if input length exceeds passed-in param length, throw new Error
-  if (input.length > length)
-    throw new Error(
-      'Mximum query length exceeded, modify query or maximum permitted query length'
-    );
-  // if input depth is within the permitted limits, return input
-  return input;
+// init sanitizeQuery, an Express middleware function users will require and invoke in their applications to generate the secrets used during the creation of JWTs
```
Comment (joie-zhang) [inline sanitizeQuery.js:11]: comment correction! :)

File: sanitizeQuery.js [hunk 3]
```diff
@@ -1,85 +1,47 @@
...
+        return next({
+          log: `Error at shieldQL sanitizeQuery while reading env file: ${err}`,
+          status: 400,
+          message: 'Internal server error',
+        });
+      // init paramsArr as array with single element, input query
+      const paramsArr = [req.body.query];
+      // check if result (the .env file) contains any params for strict, maxLength, and maxDepth
+      paramsArr.push(result.strictShieldQL ? result.strictShieldQL : false);
+      paramsArr.push(result.maxLengthShieldQL ? result.maxLengthShieldQL : 10);
```
Comment (joie-zhang) [inline sanitizeQuery.js:29]: maxDepthShieldQL

File: sanitizeQuery.js [hunk 4]
```diff
@@ -1,85 +1,47 @@
...
+      paramsArr.push(result.maxDepthShieldQL ? result.maxDepthShieldQL : 2000);
```
Comment (joie-zhang) [inline sanitizeQuery.js:30]: maxLengthShieldQL

**Review by joie-zhang — APPROVED** [review]
Great work! Amazing tests and sanitizeQuery wrapper! And professionalizing the library by deleting extra files! Woohoo!!

Note to check path of shieldql.json file!

**Pull request merged**